src/servers/trigger.ts

```diff
@@ -101,7 +106,7 @@ export async function onchainTransactionEventHandler(tx) {
 
     let user
     try {
-      user = await User.findOne({ "onchain.address": tx.output_addresses })
+      user = await User.findOne({ "onchain.address": { $in: tx.output_addresses } })
```

Comment:
the previous query was not working

Reply:
mmm. I guess this loop was never executed then. 

I guess onchain receipts are validated from the trigger on new blocks. 

there is a flaw in the logic below anyway:
```
    if (tx.is_confirmed === false) {
     // ...
    } else {
     // code logic
    }
```

with `tx.confirmed`, tx is being considered confirm with 1 block finality IIRC. but we have currently by default a 2 blocks confirmation before crediting user balance. so going into 

```
      const wallet = await WalletFactory({ user, logger: onchainLogger })
      await wallet.updateOnchainReceipt()
``` 

is basically a no-op right now.

Reply:
you are right, just setup pending notification here and add confirmation after `ledger.addOnchainReceipt`